File: tests/e2e/fixtures/typescript/src/core/product-service.ts
import type { DatabaseConnection } from './database';

export interface Product {
  id: string;
  name: string;
  price: number;
  stock: number;
}

export class ProductService {
  constructor(private database: DatabaseConnection) {}

  async getProduct(id: string): Promise<Product | null> {
    return {
      id,
      name: 'Test Product',
      price: 100,
      stock: 10
    };
  }

  async createProduct(data: Omit<Product, 'id'>): Promise<Product> {
    return {
      id: '1',
      ...data
    };
  }
}

File: tests/fixtures/swift-project/Sources/Services/ProductService.swift
import Foundation

/// 產品服務
public class ProductService {
    private var products: [UUID: Product] = [:]

    public init() {}

    /// 建立產品
    public func createProduct(name: String, price: Decimal) -> Product {
        let product = Product(name: name, price: price)
        products[product.id] = product
        return product
    }

    /// 取得產品
    public func getProduct(id: UUID) -> Product? {
        return products[id]
    }

    /// 計算購物車總價
    public func calculateTotal(productIds: [UUID]) -> Decimal {
        return productIds.compactMap { products[$0]?.price }.reduce(0, +)
    }
}
